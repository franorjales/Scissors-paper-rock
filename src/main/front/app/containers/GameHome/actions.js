/*
 *
 * GameHome actions
 *
 */

import {
  PLAY_MATCH,
  RESTART_GAME,
  PUT_GAME,
  PUT_ERROR,
  PUT_HISTORICAL,
  GET_HISTORICAL,
  CHANGE_HISTORICAL_SCREEN_ON,
} from './constants';

export function playMatchAction(user) {
  return {
    type: PLAY_MATCH,
    user,
  };
}

export function restartGameAction(user) {
  return {
    type: RESTART_GAME,
    user,
  };
}

export function putGame(game) {
  return {
    type: PUT_GAME,
    game,
  };
}

export function putError(error) {
  return {
    type: PUT_ERROR,
    error,
  };
}

export function getHistoricalAction() {
  return {
    type: GET_HISTORICAL,
  };
}

export function putHistorical(historicalGames) {
  return {
    type: PUT_HISTORICAL,
    historicalGames,
  };
}

export function changeHistoricalScreenOnAction() {
  return {
    type: CHANGE_HISTORICAL_SCREEN_ON,
  };
}
