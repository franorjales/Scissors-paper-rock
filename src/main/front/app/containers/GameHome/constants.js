/*
 *
 * GameHome constants
 *
 */

export const PLAY_MATCH = 'app/GameHome/PLAY_MATCH';
export const RESTART_GAME = 'app/GameHome/RESTART_GAME';
export const PUT_GAME = 'app/GameHome/PUT_GAME';
export const PUT_ERROR = 'app/GameHome/PUT_ERROR';
export const GET_HISTORICAL = 'app/GameHome/GET_HISTORICAL';
export const PUT_HISTORICAL = 'app/GameHome/PUT_HISTORICAL';
export const CHANGE_HISTORICAL_SCREEN_ON =
  'app/GameHome/CHANGE_HISTORICAL_SCREEN_ON';
