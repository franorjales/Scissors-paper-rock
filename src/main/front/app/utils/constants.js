export const RESTART_ON_REMOUNT = '@@saga-injector/restart-on-remount';
export const DAEMON = '@@saga-injector/daemon';
export const ONCE_TILL_UNMOUNT = '@@saga-injector/once-till-unmount';

export const SCISSORS_VALUE = 1;
export const SCISSORS = 'Scissors';
export const PAPER_VALUE = 2;
export const PAPER = 'Paper';
export const ROCK_VALUE = 3;
export const ROCK = 'Rock';
export const WINNER_PLAYER_ONE_ID = 1;
export const WINNER_PLAYER_TWO_ID = 2;
export const DRAW_VALUE = 3;
export const PLAYER_ONE = 'Player one';
export const PLAYER_TWO = 'Player two';
export const DRAW = 'Draw';
