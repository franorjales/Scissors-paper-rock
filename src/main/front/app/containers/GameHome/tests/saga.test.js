/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeLatest } from 'redux-saga/effects';
import gameHomeSaga, { playMatch, restartGame, getHistorical } from '../saga';
import { RESTART_GAME, PLAY_MATCH } from '../constants';
import { putError, putGame, putHistorical } from '../actions';

const action = { user: 'testUserName' };

describe('playMatch Saga', () => {
  let getPlayMatchGenerator;

  beforeEach(() => {
    getPlayMatchGenerator = playMatch(action);

    const callDescriptor = getPlayMatchGenerator.next(action).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the putGame action if it requests the data successfully', () => {
    const response = {
      user: 'testUserName',
      matches: [],
      numberOfmatchesPlayeds: 0,
    };
    const putDescriptor = getPlayMatchGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(putGame(response)));
  });

  it('should call the putError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getPlayMatchGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(putError(response)));
  });
});

describe('restartGame Saga', () => {
  let getRestartGameGenerator;

  beforeEach(() => {
    getRestartGameGenerator = restartGame(action);

    const callDescriptor = getRestartGameGenerator.next(action).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the putGame action if it requests the data successfully', () => {
    const response = {
      user: 'testUserName',
      matches: [],
      numberOfmatchesPlayeds: 0,
    };
    const putDescriptor = getRestartGameGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(putGame(response)));
  });

  it('should call the putError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getRestartGameGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(putError(response)));
  });
});

describe('getHistorical Saga', () => {
  let getHistoricalGenerator;

  beforeEach(() => {
    getHistoricalGenerator = getHistorical(action);

    const callDescriptor = getHistoricalGenerator.next(action).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the putGame action if it requests the data successfully', () => {
    const response = {
      totalRoundsPlayed: 1,
      totalWinsForPlayerOne: 1,
      totalWinsForPlayerTwo: 1,
      totalDraws: 1,
    };
    const putDescriptor = getHistoricalGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(putHistorical(response)));
  });

  it('should call the putError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getHistoricalGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(putError(response)));
  });
});

describe('gameHomeSaga Saga', () => {
  const gameHomeSagaTest = gameHomeSaga();

  it('should start task to watch for actions', () => {
    let takeLatestDescriptor = gameHomeSagaTest.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(PLAY_MATCH, playMatch));
    takeLatestDescriptor = gameHomeSagaTest.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(RESTART_GAME, restartGame));
  });
});
