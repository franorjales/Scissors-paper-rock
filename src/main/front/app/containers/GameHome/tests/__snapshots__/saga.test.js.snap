// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHistorical Saga should call the putError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/getHistoricalGames",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getHistorical Saga should dispatch the putGame action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/getHistoricalGames",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`playMatch Saga should call the putError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/playMatch?user=testUserName",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`playMatch Saga should dispatch the putGame action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/playMatch?user=testUserName",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`restartGame Saga should call the putError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/restartGame?user=testUserName",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`restartGame Saga should dispatch the putGame action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/api/restartGame?user=testUserName",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
